### DISCUSSION TITLE
frames per call back

### DISCUSSION THREAD

**Discussion by MohamedkhGhonim** [google/oboe#1692] [Ideas]
is there any general form to set the frame per call back to the builder I noticed that when the sample rate is 44100 or 48000 the frame per call back be 1024 and for sample rate 22050 it is a 2048, what is the appropriate number of frames per call back that can work for any sample rate ?

**Comment by philburk** [comment]
You should only set the data frames per call back when you specifically
need a particular value. An example might be when you're doing an FFT and
you need exactly 512 frames. If you don't need an exact value then it is
much better to just let Oboe decide on the callback size.

**Comment by MohamedkhGhonim** [comment]
If there is any doc or sample for that ?

**Comment by philburk** [comment]
> If there is any doc or sample for that ?

We added a doc: https://google.github.io/oboe/classoboe_1_1_audio_stream_builder.html#aef61b952f2e4cf1e0d77bcfd45dde623

Most sample code does NOT call setFramesPerDataCallback().
